pkg/controllers/dynakube/logmodule/reconciler.go

```diff
@@ -38,11 +45,23 @@ func NewReconciler(clt client.Client,
 		configSecretReconciler:           configsecret.NewReconciler(clt, apiReader, dk),
 		daemonsetReconciler:              daemonset.NewReconciler(clt, apiReader, dk),
 		oneAgentConnectionInfoReconciler: oaconnectioninfo.NewReconciler(clt, apiReader, dtc, dk),
+		monitoredEntitiesReconciler:      monitoredentities.NewReconciler(dtc, dk),
 	}
 }
 
 func (r *Reconciler) Reconcile(ctx context.Context) error {
-	err := r.oneAgentConnectionInfoReconciler.Reconcile(ctx)
+	err := r.monitoredEntitiesReconciler.Reconcile(ctx)
+	if err != nil {
+		return err
+	}
+
+	if r.dk.Status.KubernetesClusterMEID == "" || r.dk.Status.KubernetesClusterName == "" {
+		log.Info("the status of the DynaKube is missing information about the kubernetes monitored-entity, skipping logmodule deployment")
```

Comment:
I personally do not mind by not erroring here and instead just log it and return early.

Do you think we could add this case somehow to the dynakube status with an appropriate reason?

Reply:
we should add this to the deployment status of the log module so it's obvious not only in logs but also in the Dynakube itself

Reply:
I now error instead, and you will see in the logs: (it's a bit loud)
```
{"level":"error","ts":"2024-10-17T07:51:01.989Z","logger":"dynakube","msg":"error reconciling DynaKube","namespace":"dynatrace","name":"zib50933zib50933zib50933zib50933zib5093","error":"the status of the DynaKube is missing information about the kubernetes monitored-entity, skipping logmodule deployment","stacktrace":"github.com/Dynatrace/dynatrace-operator/pkg/controllers/dynakube.(*Controller).handleError\n\tgithub.com/Dynatrace/dynatrace-operator/pkg/controllers/dynakube/controller.go:191\ngithub.com/Dynatrace/dynatrace-operator/pkg/controllers/dynakube.(*Controller).Reconcile\n\tgithub.com/Dynatrace/dynatrace-operator/pkg/controllers/dynakube/controller.go:145\nsigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller[...]).Reconcile\n\tsigs.k8s.io/controller-runtime@v0.19.0/pkg/internal/controller/controller.go:116\nsigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller[...]).reconcileHandler\n\tsigs.k8s.io/controller-runtime@v0.19.0/pkg/internal/controller/controller.go:303\nsigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller[...]).processNextWorkItem\n\tsigs.k8s.io/controller-runtime@v0.19.0/pkg/internal/controller/controller.go:263\nsigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller[...]).Start.func2.2\n\tsigs.k8s.io/controller-runtime@v0.19.0/pkg/internal/controller/controller.go:224"}
{"level":"info","ts":"2024-10-17T07:51:01.989Z","logger":"dynakube","msg":"reconciling DynaKube finished","namespace":"dynatrace","name":"zib50933zib50933zib50933zib50933zib5093","result":{"Requeue":false,"RequeueAfter":0}}
{"level":"error","ts":"2024-10-17T07:51:01.990Z","logger":"main","msg":"Reconciler error","controller":"dynakube-controller","controllerGroup":"dynatrace.com","controllerKind":"DynaKube","DynaKube":{"name":"zib50933zib50933zib50933zib50933zib5093","namespace":"dynatrace"},"namespace":"dynatrace","name":"zib50933zib50933zib50933zib50933zib5093","reconcileID":"cc5891b6-94e1-4c21-8298-0c8b432ded14","error":"the status of the DynaKube is missing information about the kubernetes monitored-entity, skipping logmodule deployment","stacktrace":"sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller[...]).reconcileHandler\n\tsigs.k8s.io/controller-runtime@v0.19.0/pkg/internal/controller/controller.go:316\nsigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller[...]).processNextWorkItem\n\tsigs.k8s.io/controller-runtime@v0.19.0/pkg/internal/controller/controller.go:263\nsigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller[...]).Start.func2.2\n\tsigs.k8s.io/controller-runtime@v0.19.0/pkg/internal/controller/controller.go:224"}

```

Also the `Phase` of the `DynaKube` will be in `Error`

Is this what we want? (I kinda want this 😅 )